rust/src/array.rs

```diff
@@ -239,6 +241,67 @@ impl<T: ArrowNumericType> PrimitiveArray<T> {
     }
 }
 
+macro_rules! def_numeric_math_ops {
+    ( $ty:ident, $native_ty:ident ) => {
+        impl PrimitiveArray<$ty> {
+            fn math_op<F>(self, other: &PrimitiveArray<$ty>, op: F) -> Result<PrimitiveArray<$ty>>
```

Comment:
I'm thinking whether we should expose some basic functions on primitive array, such as `map`, `fold`, etc. The `math_op` can be generalized to accept any function that takes two arrays. In functional world, it can be achieved via `zip` plus `map`, but I'm not sure how to implement `zip` on multiple primitive arrays..